modules/programs/vivid.nix

```diff
@@ -34,8 +37,11 @@ in
     enableFishIntegration = mkFishIntegrationOption { inherit config; };
 
     colorMode = mkOption {
-      type = with types; nullOr str;
-      default = null;
+      type = types.enum [
+        "8-bit"
+        "24-bit"
+      ];
+      default = "24-bit";
```

Comment:
Not a fan of changes that add maintenance burden for no real benefit. Also forces a cli flag instead of just going with the true app default.

ie: If they ever added different bit supports or changed their default, we'd be enforcing an enum that wouldn't allow a user to change and / or we'd enforce a default that conflicts with upstream.

Reply:
If you feel strong enough about wanting the enum back (assuming for documentation / guiding a user), I'd prefer to keep the `nullOr` for backwards compatibility and to decide whether to pass the cli flag at least and allow using upstream default. Something we've done elsewhere for maximum flexibility is a `nullOr (either str enum[])` to guide towards known good values but allow someone to pass a value that could be acceptable in the future.

Reply:
I will implement it tomorrow then!
And thanks for the hint about `nullOr (either str enum[])`, wasn't aware of it, but I will keep it in mind for the future!
So a potential implementation would look like this:
```nix
type =
  with types;
  nullOr (either str enum [
    "8-bit"
    "24-bit"
  ]);
```

Reply:
> Something we've done elsewhere for maximum flexibility is a `nullOr (either str enum[])` to guide towards known good values but allow someone to pass a value that could be acceptable in the future.

Thanks for sharing! This is a great pattern.